### PR TITLE
[Blocked] Update faas version to 0.7.7 in function templates and docs

### PR DESCRIPTION
**Pull request by iyovcheva**
Signed-off-by: Ivana Yovcheva (VMware) <iyovcheva@vmware.com>

Keeping documentation and function templates up-to-date with the latest faas version.

## Description

A new script is available to automate the updates.

To use it run
`./contrib/replace_in_files.sh <old_v> <new_v>`,
in this case
`./contrib/replace_in_files.sh 0.7.1 0.7.7`

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Raised by @alexellis 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: update_version_templates.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+OLD_VERSION=$1
+NEW_VERSION=$2
+
+echo "Substituting templates for version $OLD_VERSION with $NEW_VERSION"
+
+grep -r -l --fixed-strings $OLD_VERSION .*.md * | while read fname; do
+    echo "Replasing version in file $fname"
```
Comment (alexellis) [inline update_version_templates.sh]: Typo in this line

**Comment by iyovcheva** [comment]
Derek add reviewer: alexellis

File: update_version_templates.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
Comment (alexellis) [inline update_version_templates.sh]: Can this be put in the contrib folder please?

Let's call it replace_in_files.sh instead?

File: update_version_templates.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+OLD_VERSION=$1
+NEW_VERSION=$2
+
+echo "Substituting templates for version $OLD_VERSION with $NEW_VERSION"
```
Comment (alexellis) [inline update_version_templates.sh]: Substituting-> Replacing

**Comment by alexellis** [comment]
What a smart idea 😄 I like it.. have left a couple of comments for you.

File: prometheus/Dockerfile.alertmanager [hunk 4]
```diff
@@ -1,3 +1,3 @@
-FROM prom/alertmanager:v0.7.1
+FROM prom/alertmanager:v0.7.7
```
Comment (alexellis) [inline prometheus/Dockerfile.alertmanager]: This shouldn't have changed

**Comment by alexellis** [comment]
@iyovcheva too much changed during your search/replace. The only files that should have changed were the sample functions.

Check out the change to Prometheus and other unrelated files to the watchdog version. I think we may need to revert this PR / commit and attempt to use it again with a more controlled scope.

**Comment by alexellis** [comment]
I'm going to close this PR and suggest you make a fresh attempt at this without the global search and replace script. It looks like a smart idea but it's updated files it shouldn't have done and I'd worry that other people would do this too.

**Comment by iyovcheva** [comment]
Testing derek on PRs

**Comment by iyovcheva** [comment]
Derek close

**Comment by iyovcheva** [comment]
Derek open

**Comment by iyovcheva** [comment]
Derek reopen

**Comment by alexellis** [comment]
Derek close: please can we start again with a new PR/branch and only run any global search / replace in the samples folder?

**Comment by alexellis** [comment]
Go to the latest watchdog version we have please.

**Comment by alexellis** [comment]
For a new PR use that latest watchdog version whichever that is.

**Pull request closed without merging**